provider/internal/cache.go

```diff
@@ -470,8 +470,7 @@ func (c CacheFrom) validate(preview bool) (*controllerapi.CacheOptionsEntry, err
 	if len(parsed) == 0 {
 		// This can happen for example if we have a GHA cache but no GitHub
 		// environment variables set.
-		// Shouldn't happen...
-		return nil, nil
+		return nil, errors.New("cacheFrom not compatible with the current environment")
```

Comment:
Could the error message give users a hint what the problem is? "not compatible" is not actionable.

Reply:
I wonder if this is a breaking change. When running the build with `exec: false` (the default) this doesn't seem to panic or error out for me but I think it would after this change, right?

Reply:
This error would be triggered if there's a "cacheFrom" entry of type gha, and it isn't well-formed. I suppose it is true that the configuration need not be valid if exec is false, and I'll double-check that the validation is skipped.

I'm also concerned about the preview mode behavior, since apparently the code sees zero values as opposed to unknowns.

Reply:
Nice! 🚀